x-pack/plugins/case/server/plugin.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { first, map } from 'rxjs/operators';
+import { CoreSetup, Logger, PluginInitializerContext } from 'kibana/server';
+import { ConfigType } from './config';
+import { initCaseApi } from './routes/api';
+import { CaseService } from './services';
+import { PluginSetupContract as SecurityPluginSetup } from '../../security/server';
+
+function createConfig$(context: PluginInitializerContext) {
+  return context.config.create<ConfigType>().pipe(map(config => config));
+}
+
+export interface PluginsSetup {
+  security: SecurityPluginSetup;
+}
+
+export class CasePlugin {
+  private readonly log: Logger;
+
+  constructor(private readonly initializerContext: PluginInitializerContext) {
+    this.log = this.initializerContext.logger.get();
+  }
+
+  public async setup(core: CoreSetup, plugins: PluginsSetup) {
+    const config = await createConfig$(this.initializerContext)
+      .pipe(first())
+      .toPromise();
+
+    if (!config.enabled) {
+      return;
+    }
+    const service = new CaseService(this.log);
+
+    this.log.debug(
+      `Setting up Case Workflow with core contract [${Object.keys(
+        core
+      )}] and plugins [${Object.keys(plugins)}]`
+    );
+
+    const caseService = await service.setup({
+      authentication: plugins.security ? plugins.security.authc : null,
```

Comment:
You've [marked `security` as a required plugin](https://github.com/stephmilovic/kibana/blob/1172e1be6f5aafa65d6fa93ea37009205dd3fb9e/x-pack/plugins/case/kibana.json#L5), so you're guaranteed to get the plugins's setup contract here. If the Kibana security plugin is disabled, then yours will never run. 

If security is an optional dependency, then we should update kibana.json and your PluginsSetup interface to mark it as such.